tests/SchematronValidatorTest.php

```diff
@@ -50,10 +50,15 @@ public function it_fails_on_invalid(string $format) : void
             $this->assertEquals(
                 [
                     new Failure(
-                        'Must not be empty',
+                        'Element must not be empty',
                         6,
                         $xpath->query('/example:parent/example:child')->item(0)
                     ),
+                    new Failure(
+                        'Attribute must not be empty',
+                        6,
+                        $xpath->query('/example:parent/example:child/@attribute')->item(0)
```

Comment:
Is this enough context? I guess the `DOMNode` contained in `Failure` has some information about what element it was on?

Reply:
Yep, since it's retrieved from the document.